p2p/server.py

```diff
@@ -122,36 +159,109 @@ def __init__(self,
         )
 
         # Create and register peer in peer_pool
-        eth_peer = ETHPeer(
-            remote=initiator_remote, privkey=self.privkey, reader=reader,
-            writer=writer, aes_secret=aes_secret, mac_secret=mac_secret,
-            egress_mac=egress_mac, ingress_mac=ingress_mac, chaindb=self.chaindb,
+        peer = self.peer_class(
+            remote=initiator_remote,
+            privkey=self.privkey,
+            reader=reader,
+            writer=writer,
+            aes_secret=aes_secret,
+            mac_secret=mac_secret,
+            egress_mac=egress_mac,
+            ingress_mac=ingress_mac,
+            chaindb=self.chaindb,
             network_id=self.network_id
         )
-        self.peer_pool.add_peer(eth_peer)
-
-
-def decode_authentication(ciphertext: bytes,
-                          privkey: datatypes.PrivateKey
-                          ) -> Tuple[datatypes.PublicKey, bytes, datatypes.PublicKey]:
-    """
-    Decrypts and decodes the ciphertext msg.
-    Returns the initiator's ephemeral pubkey, nonce, and pubkey.
-    """
-    if len(ciphertext) < ENCRYPTED_AUTH_MSG_LEN:
-        raise ValueError("Auth msg too short: {}".format(len(ciphertext)))
-    elif len(ciphertext) == ENCRYPTED_AUTH_MSG_LEN:
-        sig, initiator_pubkey, initiator_nonce, _ = decode_auth_plain(
-            ciphertext, privkey)
+
+        await self.do_p2p_handshake(peer)
+
+    async def do_p2p_handshake(self, peer: BasePeer) -> None:
+        try:
+            # P2P Handshake.
+            await peer.do_p2p_handshake(),
+        except (HandshakeFailure, asyncio.TimeoutError) as e:
+            self.logger.debug('Unable to finish P2P handshake: %s', str(e))
+        else:
+            # Run peer and add peer.
+            self.peer_pool.start_peer(peer)
+
+
+def _test() -> None:
```

Comment:
It'd be nice to actually have these run in our CI environment.  Thoughts on adding that as an extra CI run?  Alternatively, it seems like we should move away from this pattern since this code ends up rotting since it's not covered by CI.

@gsalgado you have a better handle on how these are used and how critical they are.  Do you think we can phase them out?  Roll them into CI?  Move them to a `scripts` directory of some sort with some basic tests to demonstrate they *probabalistically* work?

Reply:
No objection to running in CI.
And also, I think it makes sense if we have some scripts for `trinity` daemon?

Reply:
I understand your concern but most of those (chain.py, state.py, sync.py) need to run against a fully-synced geth instance and it'd be really tricky to try and somehow automate checks to ensure they work. Also, and I know this is not a good argument, this is what I use (several times a day, actually) to manually test all the p2p code, so it is unlikely to bitrot in the near future

As for moving them to a `scripts` dir, I think that may be a good idea as it'd force us to refactor some of this into helpers to reduce duplication, but at the same time I like the fact that now they live right next to the code they exercise, so I can just run, say, `python -m p2p.chain` to test the chain syncer

Reply:
@gsalgado since you are the primary consumer of these I'm going to shelve this idea for now since they are clearly working for you and I don't want to fix what isn't broken.  Once we're in public release mode we can try and address this.